src/platform_impl/android/mod.rs

```diff
@@ -502,15 +502,16 @@ impl<T: 'static> EventLoop<T> {
                         let pointers: Box<
                             dyn Iterator<Item = android_activity::input::Pointer<'_>>,
                         > = match phase {
-                            event::TouchPhase::Started
-                            | event::TouchPhase::Ended => {
+                            event::TouchPhase::Started => {
                                 Box::new(
                                     std::iter::once(motion_event.pointer_at_index(
-                                        motion_event.pointer_index(),
+                                        // In case it could be multi-touch gestures this requires the index of the last pointer.
+                                        motion_event.pointer_count() - 1_usize,
                                     ))
                                 )
                             },
-                            event::TouchPhase::Moved
+                            event::TouchPhase::Ended
```

Comment:
All `Up/PointerUp/Down/PointerDown` events should only relate to a single pointer as far as I know so it doesn't look right for this to move.

Overall there may be multiple active pointers during a multi-touch gesture but the `Ended` action itself should only relate to a single pointer.

Since we combine `Up` and `PointerUp` into `Ended` this change implies to me that if you were to start a multi-touch guesture with 4 fingers and then lift any finger that would result in an `Ended` event for _all_ the fingers/pointers because we'd iterate all active pointers and put them in the `Ended` phase.

Reply:
Thank you for your clarification, I see this problem now.